### PR TITLE
Allow the rotation of tokens

### PR DESCRIPTION
**Pull request by rullzer**
This for example will allow rotating the apptoken for oauth

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: lib/private/Authentication/Token/IProvider.php [hunk 1]
```diff
@@ -145,4 +145,14 @@ public function getPassword(IToken $token, string $tokenId): string;
...
+
+	/**
+	 * Rotate the token. Usefull for for example oauth tokens
+	 *
+	 * @param IToken $token
+	 * @param string $oldTokenId
+	 * @param string $newTokenId
+	 * @return IToken
+	 */
+	public function setNewToken(IToken $token, string $oldTokenId, string $newTokenId): IToken;
```
Comment (ChristophWurst) [inline lib/private/Authentication/Token/IProvider.php]: Would it make sense to name the method `rotate` or similar?

File: lib/private/Authentication/Token/IToken.php [hunk 2]
```diff
@@ -96,7 +96,30 @@ public function getScopeAsArray(): array;
 	 */
 	public function setScope($scope);
 
+	/**
+	 * Get the name of the token
+	 * @return string
```
Comment (ChristophWurst) [inline lib/private/Authentication/Token/IToken.php]: nitpick for future PRs: return type annotations are not necessary here :wink:

**Review by ChristophWurst — COMMENTED** [review]
Looks good. Found one minor naming issue.

File: lib/private/Authentication/Token/IProvider.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Authentication/Token/IProvider.php]: Yes will do. To much token in this indeed

**Pull request merged**